apps/outlook-rooms-addin/src/tests/map.service.spec.ts

```diff
@@ -0,0 +1,175 @@
+            BrowserModule,
+            BrowserAnimationsModule,
+            MatBottomSheetModule,
+            ComponentsModule,
+        ],
+        providers: [
+            {
+                provide: MatBottomSheet,
+                useValue: {
+                    open: jest.fn((RoomTileComponent) => {}),
+                    afterDismissed: jest.fn(),
+                },
+            },
+            {
+                provide: RoomConfirmService,
+                useClass: roomConfirmServiceStub,
+            },
+        ],
+        declarations: [
+            MockComponent(RoomTileComponent),
+            MockComponent(MapPinComponent),
+        ],
+    });
+
+    beforeEach(() => {
+        spectator = createService();
+        jest.clearAllMocks();
+        jest.resetModules();
+        ngMocks.reset();
+    });
+
+    it('should create service', () => {
+        expect(spectator.service).toBeTruthy();
+    });
+
+    it('should return a selected space', async () => {
+        const room_service = spectator.inject(RoomConfirmService);
+        room_service.selected_space$ = of(mockSpace);
+
+        expect(JSON.stringify(spectator.service.selected_space$)).toBe(
+            JSON.stringify(room_service.selected_space$)
+        );
+    });
+    it('should return a list of map IDs with no duplicates', async () => {
+        const available_spaces = of([
+            mockSpace,
+            mockSpace,
+            mockSpace,
+            mockSpace,
+        ]);
+        let maps_list;
+        const map_list_spy = jest.spyOn(spectator.service, 'loadMap');
+        await spectator.service.locateSpaces(available_spaces);
+        spectator.service.maps_list$.subscribe((list) => {
+            maps_list = list;
+        });
+
+        expect(maps_list.length).toBe(1); //3 spaces with the same map_ID were passed through, only 1 should be returned
+        expect(map_list_spy).toHaveBeenCalled();
+    });
+    it('should have a flag indicating whether the map has loaded', async () => {
+        let flag;
+        await spectator.service.map_loaded$.pipe(take(1)).toPromise();
+        spectator.service.map_loaded$.subscribe((value) => (flag = value));
+        expect(flag).toBe(false);
+
+        await spectator.service.loadMap();
+        expect(flag).toBe(true);
+    });
+
+    it('should create map actions for all available spaces', async () => {
+        let map_actions;
+        let spaces_count;
+        const available_spaces = of([
+            mockSpace,
+            mockSpace,
+            mockSpace,
+            mockSpace,
+        ]);
+
+        await spectator.service.locateSpaces(available_spaces);
+
+        spectator.service.map_actions$.subscribe((actions) => {
+            map_actions = actions;
+        });
+        available_spaces.subscribe((spaces) => {
+            spaces_count = spaces;
+        });
+
+        expect(map_actions.length).toBe(spaces_count.length);
+
+        map_actions.forEach((map) => {
```

Comment:
Might be better to change this to an `every` call.
e.g. `expect(map_action.every({ action } => action === 'click')).toBeTruthy()`